File: src/domain/enums/CoachContentType.ts
/**
 * Kind of coach-authored content (EM10). Mirrors the backend
 * `com.musclemap.coach.CoachContentType`; the string values match the persisted
 * enum names so the same value travels UI → API → DB unchanged.
 */
export enum CoachContentType {
  /** A demonstration of how to perform a single exercise (the default). */
  Technique = 'TECHNIQUE',
  /** An educational lesson: form cues, anatomy, programming theory. */
  Education = 'EDUCATION',
  /** A full training program published for members to follow. */
  Program = 'PROGRAM',
}

File: backend/src/main/java/com/musclemap/coach/CoachContentType.java
package com.musclemap.coach;

/**
 * Kind of coach-authored content (EM10). Persisted as its {@code name()} (see the
 * {@code coach_videos.content_type} CHECK constraint), so the application enum and
 * the DB stay in lockstep without a Postgres ENUM type.
 */
public enum CoachContentType {
    /** A demonstration of how to perform a single exercise (the default). */
    TECHNIQUE,
    /** An educational lesson: form cues, anatomy, programming theory. */
    EDUCATION,
    /** A full training program the coach publishes for members to follow. */
    PROGRAM
}
